addons/website/static/src/js/tours/tour_utils.js

```diff
@@ -166,6 +167,23 @@ function clickOnElement(elementName, selector) {
     };
 }
 
+/**
+ * Click on the top right edit button and wait for the edit mode
+ *
+ * @param {string} position Where the purple arrow will show up
+ */
+function clickOnEditAndWaitEditMode(position = "bottom") {
+    return [{
+        content: _t("<b>Click Edit</b> to start designing your homepage."),
+        trigger: ".o_menu_systray .o_edit_website_container a",
+        position: position,
+    }, {
+        content: "Check that we are in edit mode",
+        trigger: '#oe_snippets.o_loaded',
```

Comment:
The selector your are referring in your commit message about `registerWebsitePreviewTour` is actually using `'.o_website_preview.editor_enable.editor_has_snippets'`. Those should be the same, they are set at the same time from the same code.
I would use the same here too to remain consistant, but nitpicking.
And you can also actually grep `editor_has_snippets` and not just `o_loaded` -> some tour also did the same check after click on edit which need to be removed to, cf `full_screen_web_editor` in website_slides (you edited the file to edit the new tool but did not remove that step)